### ISSUE TITLE
[Package Issue]: Microsoft.DotNet.Runtime.3_1

### ISSUE THREAD

**Issue by Egon099** [microsoft/winget-pkgs#157330]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

No applicable update found.

### Brief description of your issue

Microsoft .NET Core Runtime - 2.0.9 (x64)
Microsoft .NET Core Runtime - 2.2.8 (x64)
Microsoft .NET Core 1.1.13 - Runtime (x86)
Microsoft .NET Core 1.1.13 - Runtime (x64)
Microsoft .NET Core Runtime - 2.2.8 (x86)
Microsoft .NET Core 1.0.16 - Runtime (x64)
Microsoft .NET Core 1.0.16 - Runtime (x86)
Microsoft .NET Core Runtime - 2.0.9 (x86)
Microsoft .NET Core Runtime - 2.1.30 (x64)
Microsoft .NET Core Runtime - 2.1.30 (x86)
All are id'd same Microsoft.DotNet.Runtime.3_1 though they are different programs. So always in the list as update even tho have latest versions. 

### Steps to reproduce

install older .net core and run winget update

### Actual behavior

Identifies wrong, tries to download and run update even tho nothing to update

### Expected behavior

That they would be id-d differently and no updates will be offered since they are legacy software

### Environment

```raw
Windows Package Manager v1.7.11261
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22631.3672
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.22.11261.0

Winget Directories
-----------------------------------------------------------------------------------------------------------------------
Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\Diag…
User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\sett…
Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portable Package Root              C:\Program Files\WinGet\Packages
Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages
Installer Downloads                %USERPROFILE%\Downloads

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Disabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Disabled
LocalArchiveMalwareScanOverride           Disabled
```

### Screenshots and Logs

![image](https://github.com/microsoft/winget-pkgs/assets/95533676/ab23d3ff-42c6-4998-b0a0-01d2e5437864)

**Comment by MagicAndre1981** [comment]
they are all old and out of support, so unintall them via Control Panel->Programs & Features

**Comment by Egon099** [comment]
Tried installing all .NET Core/SDK out of curiosity. All latest versions. Seems something is messed up but don't know if it is packages problem or Winget problem. Doesn't seem to be as intended?
![image](https://github.com/microsoft/winget-pkgs/assets/95533676/146e20df-1e69-48cc-9965-0cf81c1761ba)

**Comment by Egon099** [comment]
> they are all old and out of support, so unintall them via Control Panel->Programs & Features

Oh I know that. But sometimes I play old games with extension apps that require old net versions. Otherwise wouldn't have noticed or installed

**Comment by MagicAndre1981** [comment]
some older VC++ runtimes, yes, but not those old .net core packages.

**Comment by stephengillie** [comment]
These seem like they can all be installed side-by-side at the same time, so should probably each have their own `PackageIdentifier`.

**Comment by Egon099** [comment]
Yeah, they are tecnically different programs [https://dotnet.microsoft.com/en-us/download/dotnet](https://dotnet.microsoft.com/en-us/download/dotnet)
So there should be at least 12 different PackageIdentifiers.
